test/com/google/javascript/jscomp/integration/ProductionCoverageInstrumentationPassIntegrationTest.java

```diff
@@ -340,51 +272,85 @@ public void testInstrumentationMappingIsCreated() {
         .isEqualTo("[foo]");
     assertWithMessage("FileNames in the parameter mapping are not properly set")
         .that(paramMap.get(" FileNames"))
-        .isEqualTo("[i1.js]");
+        .isEqualTo("[i0.js]");
     assertWithMessage("Types in the parameter mapping are not properly set")
         .that(paramMap.get(" Types"))
         .isEqualTo("[FUNCTION]");
     assertWithMessage("Array index encoding is not performed properly")
         .that(paramMap.get("C"))
-        .isEqualTo("AAA");
+        .isEqualTo("AAACA");
   }
 
-  private String getInstrumentCodeLine(String encodedParam, int lineNo, int colNo) {
-    return Strings.lenientFormat(
-        "%s(\"%s\", %s, %s)",
-        "module$exports$instrument$code.instrumentCodeInstance.instrumentCode",
-        encodedParam,
-        lineNo,
-        colNo);
+  @Test
+  public void testGlobalArrayIsDeclared() {
+    CompilerOptions options = createCompilerOptions();
+
+    String source = lines("function foo (x) {", "console.log(x+5);", "};");
+    String useGlobalArray = lines("function ist(){", "console.log(ist_array)", "}");
+
+    String expected =
+        lines("function foo (x) {", getInstrumentCodeLine("C"), ";", "console.log( x + 5);", "};");
+
+    String expectedUseGlobalArray =
+        lines("function ist(){", getInstrumentCodeLine("E"), ";", "console.log(ist_array);", "}");
+
+    String[] sourceArr = new String[] {source, useGlobalArray};
+    String[] expectedArr =
+        new String[] {instrumentCodeExpected.concat(expected), expectedUseGlobalArray};
+
+    test(options, sourceArr, expectedArr);
   }
 
-  @Override
-  protected Compiler compile(CompilerOptions options, String[] original) {
-    Compiler compiler =
-        useNoninjectingCompiler
-            ? new NoninjectingCompiler(new BlackHoleErrorManager())
-            : new Compiler(new BlackHoleErrorManager());
+  @Test
+  public void testInstrumentationWithAdvancedOpt() {
+    CompilerOptions options = createCompilerOptions();
 
-    lastCompiler = compiler;
+    String blahExtern =
+        lines(
+            "/**",
+            "* @externs",
+            "*/",
+            "/**",
+            "*@type {!Array<string>}",
+            "*/",
+            "let ist_arr;"
+            );
+    externs = ImmutableList.of(
+        new TestExternsBuilder()
+            .addArray()
+            .addAlert()
+            .addExtra(
+                blahExtern)
+            .buildExternsFile("externs"));
+
+
+    String source = lines("function foo (x) {", "if(x) { alert(x+5); }", "} foo(true); foo(false);");
 
-    List<SourceFile> inputs = new ArrayList<>();
+    String expected =
+        lines(
+            "function a(b) {",
+            getInstrumentCodeLine("G"),";",
+            "b ? (", getInstrumentCodeLine("C"),",alert(b+5)) : ", getInstrumentCodeLine("E"),";",
```

Comment:
formatting looks off here

Reply:
Not sure what you were specifically referring too, but I assumed it was the `,alert(`? The formatter also puts most of these strings on their own line which is updated.